executor/projection.go

```diff
@@ -166,6 +173,8 @@ func (e *ProjectionExec) isUnparallelExec() bool {
 }
 
 func (e *ProjectionExec) unParallelExecute(ctx context.Context, chk *chunk.Chunk) error {
+	// transmit the requiredRows
+	e.childResult.SetRequiredRows(chk.RequiredRows(), e.maxChunkSize)
```

Comment:
should we also set the child required rows for parallel execution mechanism?

Reply:
Yep, it already works on parallel mode by introducing a new variable `requiredRowsNow `.
It used in line 309-310.